xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -477,8 +477,11 @@ private <T extends ResourceUpdate> void handleResourceUpdate(XdsResourceType.Arg
       }
 
       // For State of the World services, notify watchers when their watched resource is missing
-      // from the ADS update.
-      subscriber.onAbsent();
+      // from the ADS update. Note that we can only do this if the resource update is coming from
+      // the same xDS server that the ResourceSubscriber is subscribed to.
+      if (subscriber.serverInfo.target().equals(args.serverInfo.target())) {
```

Comment:
I don't think `target()` is specific enough. We dedup streams based on the entire serverInfo. I think we should check if the serverInfos are equal.

Reply:
I'd say it could be concerning, but we're already using it for `serverChannelMap`. So if there's a problem, it doesn't impact just this line.